File: custom-exception/src/main/java/dev/trile/customexception/SalaryTransfer.java
package dev.trile.customexception;

public class SalaryTransfer {

  private final MoneyTransfer moneyTransfer = new MoneyTransfer();
  private final SalaryCalculator salaryCalculator = new SalaryCalculator();

  public void salaryTransfer(int workingDay, int salaryPerDay, String accountNo) throws SalaryException {

    int salary = salaryCalculator.calcSalary(workingDay, salaryPerDay);
    moneyTransfer.transferMoney(accountNo, salary);
    System.out.println("Transfer Salary success");

  }

  public static void main(String[] args) {
    SalaryTransfer salaryTransfer = new SalaryTransfer();

    try {
      salaryTransfer.salaryTransfer(Integer.parseInt(args[0]), Integer.parseInt(args[1]), args[2]);
    } catch (NumberFormatException e) {
      System.out.println("Parse int error " + e.getMessage());
    } catch (SalaryException e) {
      System.out.println(e.getMessage());
    }
  }
}

File: ioc-di/src/main/kotlin/dev/trile/iocdi/SalaryTransfer.kt
package dev.trile.iocdi

import org.springframework.boot.CommandLineRunner
import org.springframework.stereotype.Component

@Component
class SalaryTransfer(
  private val moneyTransfer: MoneyTransfer,
  private val salaryCalculator: SalaryCalculator
) : CommandLineRunner {


  fun salaryTransfer(workingDay: Int, salaryPerDay: Int, accountNo: String?) {
    val salary = salaryCalculator.calcSalary(workingDay, salaryPerDay)
    moneyTransfer.transferMoney(accountNo!!, salary)
    println("Transfer Salary success")
  }

  override fun run(vararg args: String) {

    try {
      salaryTransfer(args[0].toInt(), args[1].toInt(), args[2])
    } catch (e: NumberFormatException) {
      println("Parse int error " + e.message)
    } catch (e: SalaryException) {
      println(e.message)
    }
  }

}

